File: Năm 2/Kì 2 đợt 2/ThuatToanATTT/.vscode/Phần 1/bai2.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

int isPrime(unsigned long long x){
    for(unsigned long long i = 2; i*i <= x; i++){
        if (x%i == 0) return 0;
    }
    return 1;  
}
int main(){
    int N;
    printf("Nhap N: ");
    scanf("%d", &N);
    int count = 0;
    for(unsigned long long i = pow(10,N-1); i <= pow(10,N); i++){
        if(isPrime(i)){
            count++;
            printf("%-15llu", i);
        }
        if (count == 5){
            count = 0;

        }
    }

}

File: Năm 2/Kì 2 đợt 2/ThuatToanATTT/.vscode/Phần 1/bai2.py
import math
def isPrime(n):
    if n < 2: 
        return 0
    if n == 2: 
        return 1
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return 0
    return 1
if __name__ == "__main__":
    n = int(input("Nhap n:"))
    print(f"Cac so nguyen to co {n} chu so la: ")
    count = 0
    for i in range(10**(n-1), 10**n):
        if isPrime(i):
            count+=1
            print(f"{i:-11d}", end = "")
            if count == 5:
                count = 0
                print("")